### PR TITLE
Governance Support

### PR DESCRIPTION
**Pull request by qmjava**
This PR will add governance rule in .NET middleware
#183200518

1. added governance model and helper
2. blocked traffic based on governance rule
3. used volatile instead for concurrency
4. used event notification to reduce API calls
5. convert sync calls to async calls
6. added unit tests for governance rule
7. bumped up version

Pull Request review Checklist:
 - [x] `Follows` guidelines for `public` repo ?
 - [x] `README` updated, if required ?
 - [x] `Release Version` updated ?
 - [x] `User Agent` Version updated ?
 - [x] `Includes` tests conducted and `test result` in the ticket and/or QA doc ?
 - [x] `Does NOT` include code from others?  (Unintended consequence of a bad merge) ?

### CODE REVIEW

File: Moesif.NetCore.Test/Moesif.NetCore.Test.csproj [hunk 1]
```diff
@@ -13,9 +13,13 @@
...
     <PackageReference Include="Microsoft.NET.Test.Sdk" Version="16.2.0" />
     <PackageReference Include="Microsoft.AspNetCore.TestHost" Version="2.2.0" />
+    <PackageReference Include="Moesif.Api" Version="2.0.4" />
   </ItemGroup>
 
   <ItemGroup>
     <ProjectReference Include="..\Moesif.Middleware\Moesif.Middleware.csproj" />
   </ItemGroup>
+  <ItemGroup>
+    <None Remove="Moesif.Api" />
```
Comment (praves77) [inline Moesif.NetCore.Test/Moesif.NetCore.Test.csproj:23]: What is this?

**Review by praves77 — APPROVED** [review]
lftm

File: Moesif.NetCore.Test/Moesif.NetCore.Test.csproj (hunk 1 above)
Comment (qmjava) [reply Moesif.NetCore.Test/Moesif.NetCore.Test.csproj:23]: I should remove it

Comment (praves77) [reply Moesif.NetCore.Test/Moesif.NetCore.Test.csproj:23]: Ok.. Let me know after remove and running one test to confirm build and test passes.

Comment (qmjava) [reply Moesif.NetCore.Test/Moesif.NetCore.Test.csproj]: Tests run fine

**Pull request merged**